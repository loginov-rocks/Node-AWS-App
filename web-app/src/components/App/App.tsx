import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import 'components/App/App.css';
import PageProducts from 'components/pages/PageProducts/PageProducts';
import MainLayout from 'components/MainLayout/MainLayout';
import PageProductForm from 'components/pages/PageProductForm/PageProductForm';
import PageCart from 'components/pages/PageCart/PageCart';
import PageOrders from 'components/pages/PageOrders/PageOrders';
import PageOrder from 'components/pages/PageOrder/PageOrder';
import PageProductImport from 'components/pages/admin/PageProductImport/PageProductImport';
import Auth from 'utils/auth';

interface Props {
  auth: Auth;
}

function App({ auth }: Props) {
  React.useEffect(() => {
    // Parse hash string and clear if auth data found there.
    if (auth.parseQueryString(window.location.hash.substr(1))) {
      window.location.hash = '';
    }
  }, [auth]);

  return (
      <Router>
        <Switch>
          <Route path="/">
            <MainLayout>
              <Route exact path="/">
                <PageProducts/>
              </Route>
              <Route exact path={['/admin/product-form/:id', '/admin/product-form']}>
                <PageProductForm/>
              </Route>
              <Route exact path="/cart">
                <PageCart/>
              </Route>
              <Route exact path="/admin/orders">
                <PageOrders/>
              </Route>
              <Route exact path="/admin/order/:id">
                <PageOrder/>
              </Route>
              <Route exact path="/admin/products">
                <PageProductImport/>
              </Route>
            </MainLayout>
          </Route>
        </Switch>
      </Router>
  );
}

export default App;
