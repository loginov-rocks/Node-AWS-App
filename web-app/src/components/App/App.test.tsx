import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import App from 'components/App/App';

test('renders correctly', () => {
  const wrapper = createShallow()(<App />);
  expect(wrapper).toMatchSnapshot();
});
