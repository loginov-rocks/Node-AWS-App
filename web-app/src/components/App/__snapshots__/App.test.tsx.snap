// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      path="/"
    >
      <MainLayout>
        <Route
          exact={true}
          path="/"
        >
          <PageProducts />
        </Route>
        <Route
          exact={true}
          path={
            Array [
              "/admin/product-form/:id",
              "/admin/product-form",
            ]
          }
        >
          <PageProductForm />
        </Route>
        <Route
          exact={true}
          path="/cart"
        >
          <PageCart />
        </Route>
        <Route
          exact={true}
          path="/admin/orders"
        >
          <PageOrders />
        </Route>
        <Route
          exact={true}
          path="/admin/order/:id"
        >
          <PageOrder />
        </Route>
        <Route
          exact={true}
          path="/admin/products"
        >
          <PageProductImport />
        </Route>
      </MainLayout>
    </Route>
  </Switch>
</BrowserRouter>
`;
